plugins/modules/proxmox_backup_schedule.py

```diff
@@ -0,0 +1,229 @@
+    required: false
+    type: str
+  backup_id:
+    description: The backup job ID.
+    required: false
+    type: str
+  backup_action:
+    description:
+        - If V(update_vmid), the module will update backup job with new VM ID.
+        - If V(delete_vmid), the module will remove the VM ID from all backup jobs where the VM ID has existed.
+    required: true
+    type: str
+
+extends_documentation_fragment:
+  - community.general.proxmox.documentation
+  - community.general.attributes
+  - community.general.attributes.info_module
+  - community.general.proxmox.actiongroup_proxmox
+"""
+
+EXAMPLES = """
+- name: Scheduling VM Backups based on VM name.
+  proxmox_backup_schedule:
+    api_user: 'myUser@pam'
+    api_password: '*******'
+    api_host: '192.168.20.20'
+    vm_name: 'VM Name'
+    backup_id: 'backup-b2adffdc-316e'
+    backup_action: 'update_vmid'
+
+- name: Scheduling VM Backups based on VM ID.
+  proxmox_backup_schedule:
+    api_user: 'myUser@pam'
+    api_password: '*******'
+    api_host: '192.168.20.20'
+    vm_id: 'VM ID'
+    backup_id: 'backup-b2adffdc-316e'
+    backup_action: 'update_vmid'
+
+- name: Removing backup setting based on VM name.
+  proxmox_backup_schedule:
+    api_user: 'myUser@pam'
+    api_password: '*******'
+    api_host: '192.168.20.20'
+    vm_name: 'VM Name'
+    backup_action: 'delete_vmid'
+
+- name: Removing backup setting based on VM ID.
+  proxmox_backup_schedule:
+    api_user: 'myUser@pam'
+    api_password: '*******'
+    api_host: '192.168.20.20'
+    vm_id: 'VM ID'
+    backup_action: 'delete_vmid'
+"""
+
+RETURN = """
+---
+backup_schedule:
+  description:
+    - If V(update_vmid), the backup_schedule will return True after adding the VM ID to the backup job.
+    - If V(delete_vmid), the backup_schedule will return a list of backup job IDs where the VM ID has existed after removing it.
+  returned: always, but can be empty
+  type: dict
+"""
+
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+from ansible_collections.community.general.plugins.module_utils.proxmox import (
+    proxmox_auth_argument_spec, ProxmoxAnsible, HAS_PROXMOXER, PROXMOXER_IMP_ERR)
+
+
+class ProxmoxSetVMBackupAnsible(ProxmoxAnsible):
+    # Getting backup sections
+    def get_cluster_bklist(self):
+        try:
+            backupSections = self.proxmox_api.cluster.backup.get()
+        except Exception as e:
+            self.module.fail_json(msg="Getting backup sections failed: %s" % e)
+        return backupSections
+
+    def get_cluster_specific_bkjobid(self, backup_id):
+        try:
+            specificBackupID = self.proxmox_api.cluster.backup.get(backup_id)
+        except Exception as e:
+            self.module.fail_json(msg="Getting specific backup ID failed: %s" % e)
+        return specificBackupID
+
+    def set_vmid_backup(self, backup_id, bk_id_vmids):
+        try:
+            self.proxmox_api.cluster.backup.put(backup_id, vmid=bk_id_vmids)
```

Comment:
if `self.proxmox_api.cluster.backup.put()` expects `vmid` to be a string, here is the best place for you to enforce it. (As opposed to line 169 below). Dealing with the API is a concern for _this_ method, so problems related to formatting the parameters for the API should not be dealt with elsewhere in the code. `bk_id_vmids` is clearly supposed to be a list, so make it a `list` and not a `str`. If it needs conversion, make it here, not elsewhere.